app/early-access/page.tsx

```diff
@@ -0,0 +1,46 @@
+'use client'
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/navigation'
+import PrelaunchSignUpForm from '../components/main-content/PrelaunchSignUpForm';
+import SubmissionStatus from '../components/main-content/SubmissionStatusPopup';
+
+type FlowState = 'viewingForm' |'processing' | 'confirmed';
+/**
+ * Handles the flow from viewing the signup form, processing the form submission,
+ * to confirming submission success, followed by redirection to the homepage.
+ */
+export default function PreLaunchSignupFlowContainer() {
+  const [flowState, setFlowState] = useState<FlowState>('viewingForm');
+  const router = useRouter()
+
+  const handleFormSubmit = () : void => {
+    setFlowState('processing');
+  // TODO: Expand with form submission logic & refactor setTimeOut placeholder
+     setTimeout(() => setFlowState('confirmed'), 1000); // Placeholder for actual submission logic.
+    scheduleRedirectHome();
+  }
+  /**
+   * Schedules automatic redirection to homepage after a confirmation message is displayed.
+   */
+  const scheduleRedirectHome = (): void => {
+    setTimeout(() => {
+      router.push('/');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    let timeoutId: ReturnType<typeof setTimeout>;
+    if (flowState === 'confirmed') {
+      timeoutId = setTimeout(scheduleRedirectHome, 3000);
+    }
+    return () => timeoutId && clearTimeout(timeoutId);
+  }, [flowState]);
```

Comment:
eslint is flagging a missing dependency - I think we should add `scheduleRedirectHome`or, if going with my rec above,`router`

Reply:
Added it back. Tks!